### PR TITLE
Replace ceased travis-ci.org with GitHub Actions

### PR DESCRIPTION
**Pull request by kachick**
Looks like the current CI is not running. At least I can't see the result link from badges.

https://blog.travis-ci.com/2021-05-07-orgshutdown

So I have replaced to GitHub Actions with https://github.com/asdf-vm/actions/blob/171cc3570019f3a69ba10ae269ab46990936fa9f/README.md instruction except integrating `shfmt`.

* New CI testing with `3.1.2` for now.
* Integrating `shfmt` made [a lot of diff](https://github.com/kachick/asdf-ruby/runs/7450394069?check_suite_focus=true), so I have omitted it.
* Basically disabled `ShellCheck` alerting in current code to keep as-is behavior.

### CODE REVIEW

File: bin/install [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 set -euo pipefail
 
-# shellcheck source=../lib/utils.sh
+# shellcheck source=/dev/null
```
Comment (kachick) [inline bin/install:5]: https://www.shellcheck.net/wiki/SC1090
https://www.shellcheck.net/wiki/SC1091
https://github.com/koalaman/shellcheck/wiki/directive

I have tried `source-path` directive. However it still made alerts. 🤔

**Review by Stratus3D — APPROVED** [review]
Looks good to me.

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
+name: CI
+
+on:
+  workflow_dispatch:
+  pull_request:
+    paths-ignore:
+      - "**.md"
+  push:
+    branches:
+      - 'master'
```
Comment (Stratus3D) [inline .github/workflows/ci.yml:10]: I'd be in favor of running this on all branches.

**Comment by kachick** [comment]
Thank you for reviewing and merging!
